port/board/afc-v3/payload.c

```diff
@@ -313,9 +313,9 @@ void vTaskPayload( void *pvParameters )
         case PAYLOAD_STATE_FPGA_SETUP:
 #ifdef MODULE_ADN4604
             /* Configure clock switch */
-            clock_configuration();
+            if (!clock_configuration())
 #endif
-            new_state = PAYLOAD_FPGA_ON;
+                new_state = PAYLOAD_FPGA_ON;
```

Comment:
Clever syntax here. I prefer doing this in a more explicit way. Also, you should check for MMC_OK here instead of a boolean check:
```c
#ifdef MODULE_ADN4604
    if (clock_configuration() == MMC_OK) {
        new_state = PAYLOAD_FPGA_ON;
    }
#else
    new_state = PAYLOAD_FPGA_ON;
#endif
```